packages/cli/openapi-parser/src/schema/SchemaParserContext.ts

```diff
@@ -6,11 +7,20 @@ export interface SchemaParserContext {
     logger: Logger;
     DUMMY: SchemaParserContext;
 
+    sdkLanguage: generatorsYml.GenerationLanguage | undefined;
     shouldUseTitleAsName: boolean;
 
     referenceExists(ref: string): boolean;
     resolveSchemaReference(schema: OpenAPIV3.ReferenceObject): OpenAPIV3.SchemaObject;
     markSchemaAsReferencedByNonRequest(schemaId: SchemaId): void;
     markSchemaAsReferencedByRequest(schemaId: SchemaId): void;
     markReferencedByDiscriminatedUnion(schema: OpenAPIV3.ReferenceObject, discrminant: string, times: number): void;
+
+    shouldUseUndiscriminatedUnionsForDiscriminated: boolean;
+    getShouldUseUndiscriminatedUnionsForDiscriminated(): boolean;
+    storeDiscriminatedUnionMetadata(
+        schema: OpenAPIV3.ReferenceObject,
+        discrminant: string,
+        discriminantValue: string
+    ): void;
```

Comment:
thoughts on making this very specific? 
`markSchemaWithDiscriminant()`  and add docs that we will append literal values here